src/partitioning.c

```diff
@@ -317,6 +317,9 @@ resolve_function_argtype(FunctionCallInfo fcinfo)
 			/* Argument is function, so our input is its result type */
 			argtype = ((FuncExpr *) node)->funcresulttype;
 			break;
+		case T_Param:
+			argtype = ((Param *) node)->paramtype;
+			break;
 		default:
 			elog(ERROR, "unsupported expression argument node type %u", nodeTag(node));
```

Comment:
might be nice to use ts_get_node_name here but that would require adding the primnodes.h nodes to that function

Reply:
will send a following PR adding it, thanks!